src/commands/Minion/autoslay.ts

```diff
@@ -307,6 +307,20 @@ export default class extends BotCommand {
 				return masterMatch && e.monsterID === usersTask.assignedTask!.monster.id;
 			});
 
+			const allMonsters = killableMonsters.filter(m => {
+				return usersTask.assignedTask!.monsters.includes(m.id);
```

Comment:
You're checking if ANY of the possible task monsters have requirements you don't meet, instead you should just check if the efficient monster has requirements you can't meet. `ehpMonster.EfficientMonster`